frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-design/experiment-design.component.ts

```diff
@@ -101,8 +101,6 @@ export class ExperimentDesignComponent implements OnInit, OnChanges, OnDestroy {
       this.isContextChanged = false;
       this.partition.clear();
       this.condition.clear();
```

Comment:
how come we remove these? what did they do exactly?

Reply:
What do you mean? I didn't remove the 3 lines you mentioned. 
```
this.isContextChanged = false;
this.partition.clear();
this.condition.clear();
```

Reply:
i meant these lines

```
this.partition.push(this.addPartitions());
this.condition.push(this.addConditions());

Reply:
As I commented below, I believe those lines are used to add default rows.

Reply:
@zackcl Correct

Reply:
@ppratikcr7 Thank you!